iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -90,6 +90,29 @@ Viewer.prototype._buildContinuationMap = function() {
             this._ixNodeMap[id].continuations = parts;
         }
     }
+    // The document may contain ix:continuations for facts or footnotes that
+    // are not in the default target document, which we need to ignore.
+    // We can't ignore these up-front, because ix:continuation doesn't have a
+    // target attribute - we need to first resolve the continuation chain.
+    // This strips all continuation elements in the map that we haven't found
+    // the head item for.
+    for (const [id, item] of Object.entries(this._continuationMap)) {
+        if (!item.isHead && item.continuationOf === undefined) {
+            const node = this._ixNodeMap[id].wrapperNodes.first();
+            const ivids = node.data('ivid').filter(item => item != id);
+            node.data('ivid', ivids);
+            if (ivids.length == 0) {
+                node.removeClass("ixbrl-continuation").removeClass("ixbrl-element");
```

Comment:
Yes, I was having exactly the same thought as I wrote it, and the clean-up logic became more and more involved.

I was thinking that the first pass would be purely to build a continuation map, so we can identify which target document(s) a continuation corresponds to in the second pass, but it might make sense to do more, as you suggest.  I'll look into this.

Reply:
I've now done this, and the result is much cleaner.  continuations now store the ID of their head node(s) which cleans up other parts of the code.